ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala

```diff
@@ -39,33 +41,35 @@ final class ApiParticipantPruningService private (
   override def bindService(): ServerServiceDefinition =
     ParticipantPruningServiceGrpc.bindService(this, executionContext)
 
-  override def prune(request: PruneRequest): Future[PruneResponse] = {
-    val submissionIdOrErr = SubmissionId
-      .fromString(
-        if (request.submissionId.nonEmpty) request.submissionId else UUID.randomUUID().toString
-      )
-      .left
-      .map(err => ErrorFactories.invalidArgument(s"submission_id $err"))
+  override def prune(request: PruneRequest): Future[PruneResponse] =
+    withEnrichedLoggingContext(logging.submissionId(request.submissionId)) { implicit logCtx =>
+      logger.info(s"Pruning up to ${request.pruneUpTo}")
+      val submissionIdOrErr = SubmissionId
+        .fromString(
+          if (request.submissionId.nonEmpty) request.submissionId else UUID.randomUUID().toString
+        )
+        .left
+        .map(err => ErrorFactories.invalidArgument(s"submission_id $err"))
 
-    submissionIdOrErr.fold(
-      Future.failed,
-      submissionId =>
-        LoggingContext.withEnrichedLoggingContext("submissionId" -> submissionId) {
-          implicit logCtx =>
-            (for {
+      submissionIdOrErr.fold(
+        Future.failed,
+        submissionId =>
+          LoggingContext.withEnrichedLoggingContext("submissionId" -> submissionId) {
```

Comment:
It looks like the submission identifier has already been put in the logging context in the first line of this method, is it not?

Reply:
Good catch!

Reply:
Fixed.